spinalcordtoolbox/scripts/sct_maths.py

```diff
@@ -300,187 +375,189 @@ def main(argv: Sequence[str]):
-    if arguments.otsu is not None:
-        param = arguments.otsu
-        data_out = sct_math.otsu(data, param)
-
-    elif arguments.adap is not None:
-        param = arguments.adap
-        data_out = sct_math.adap(data, param[0], param[1])
-
-    elif arguments.otsu_median is not None:
-        param = arguments.otsu_median
-        data_out = sct_math.otsu_median(data, param[0], param[1])
-
-    elif arguments.thr is not None or arguments.uthr is not None:
-        data_out = sct_math.threshold(data, arguments.thr, arguments.uthr)
+    # Handle `-shape` and `-dim` for `-dilate` and `-erode`
+    if len(arguments.shape) == 0:
+        shape = 'ball'
+    elif len(arguments.shape) == 1:
+        shape = arguments.shape[0]
+    else:
+        parser.error("-shape cannot be specified more than once")
+
+    if shape in ['disk', 'square']:
+        # 2D kernels need the value of `-dim`
+        if len(arguments.dim) == 0:
+            parser.error(f"-dim is required for -shape {shape}")
+        elif len(arguments.dim) == 1:
+            dim = arguments.dim[0]
+        else:
+            parser.error("-dim cannot be specified more than once")
+    else:
+        # 3D kernels don't need `-dim`
+        if len(arguments.dim) == 0:
+            dim = None
+        else:
+            parser.error(f"-dim should not be specified for -shape {shape}")
+
+    # Check that the list of operations makes sense
+    if not arguments.todo:
+        parser.error("there must be at least one operation to perform")
+    for arg_name, _ in arguments.todo[:-1]:
+        if arg_name in ['mi', 'minorm', 'corr']:
+            parser.error(f"-{arg_name}: similarity metrics are only supported "
+                         "as the last operation to perform")
+
+    # Actually do the computations
+    im = Image(arguments.i)
+    data = im.data
+    for arg_name, arg_value in arguments.todo:
+        assert data is not None
+
+        if arg_name == "add":
+            if data.ndim == 4 and not arg_value:
+                # special case to sum a 4D volume across the t axis
+                data = np.sum(data, axis=3)
+            else:
+                try:
+                    list_data = get_data_arrays(data.shape, arg_value)
+                except ValueError as e:
+                    printv(f"ERROR: -{arg_name}: {e}", 1, 'error')
+                data += np.sum(list_data, axis=0)
 
-    elif arguments.percent is not None:
-        param = arguments.percent
-        data_out = sct_math.perc(data, param)
+        elif arg_name == "sub":
+            try:
+                list_data = get_data_arrays(data.shape, arg_value)
+            except ValueError as e:
+                printv(f"ERROR: -{arg_name}: {e}", 1, 'error')
+            data -= np.sum(list_data, axis=0)
+
+        elif arg_name == "mul":
+            if data.ndim == 4 and not arg_value:
+                # special case to multiply a 4D volume across the t axis
+                data = np.sum(data, axis=3)
```

Comment:
Oops, forgot to fix this up after copy-pasting:
```suggestion
                data = np.prod(data, axis=3)
```